File: apps/api/src/main/java/com/pitvia/api/maintenance/enums/MaintenanceCategory.java
package com.pitvia.api.maintenance.enums;

import java.util.Arrays;
import java.util.Map;
import java.util.function.Function;
import java.util.stream.Collectors;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

/**
 * 整備カテゴリを表す列挙型
 *
 * @author pitvia
 * @version 1.0
 */
@Getter
@RequiredArgsConstructor
public enum MaintenanceCategory {

    /** エンジン */
    ENGINE("エンジン"),

    /** 吸排気 */
    INTAKE_EXHAUST("吸排気"),

    /** 冷却 */
    COOLING("冷却"),

    /** 駆動系 */
    DRIVETRAIN("駆動系"),

    /** 足回り */
    SUSPENSION("足回り"),

    /** ブレーキ */
    BRAKE("ブレーキ"),

    /** 電装 */
    ELECTRICAL("電装"),

    /** 外装 */
    BODY("外装"),

    /** 内装 */
    INTERIOR("内装"),

    /** エアコン */
    AIR_CONDITIONER("エアコン"),

    /** 板金 */
    BODY_REPAIR("板金"),

    /** 洗浄 */
    CLEANING("洗浄"),

    /** その他 */
    OTHER("その他");

    /**
     * 画面表示用ラベル。
     */
    private final String label;

    /**
     * code（Enum名）をキーにした検索用Map
     */
    private static final Map<String, MaintenanceCategory> MAP = Arrays.stream(values())
            .collect(Collectors.toUnmodifiableMap(
                    MaintenanceCategory::name,
                    Function.identity()));

    /**
     * ラベルコードから整備カテゴリを取得
     *
     * @param code 整備カテゴリコード
     * @return 対応する整備カテゴリ
     * @throws IllegalArgumentException 未定義のコードが指定された場合
     */
    public static MaintenanceCategory fromCode(String code) {

        MaintenanceCategory category = MAP.get(code);

        if (category == null) {
            throw new IllegalArgumentException(
                    "Unknown maintenance category code: " + code);
        }

        return category;
    }

    /**
     * DB連携用コードを返す
     *
     * @return 整備カテゴリコード
     */
    public String getCode() {
        return name();
    }

}

File: apps/web/src/shared/constants/maintenance-category.ts
/**
 * 整備カテゴリ
 * Spring: MaintenanceCategory（maintenance.enums.MaintenanceCategory）
 */
export const MAINTENANCE_CATEGORY = {
  ENGINE: "ENGINE",
  INTAKE_EXHAUST: "INTAKE_EXHAUST",
  COOLING: "COOLING",
  DRIVETRAIN: "DRIVETRAIN",
  SUSPENSION: "SUSPENSION",
  BRAKE: "BRAKE",
  ELECTRICAL: "ELECTRICAL",
  BODY: "BODY",
  INTERIOR: "INTERIOR",
  AIR_CONDITIONER: "AIR_CONDITIONER",
  BODY_REPAIR: "BODY_REPAIR",
  CLEANING: "CLEANING",
  OTHER: "OTHER",
} as const;

export type MaintenanceCategory =
  (typeof MAINTENANCE_CATEGORY)[keyof typeof MAINTENANCE_CATEGORY];

/**
 * 整備カテゴリの表示用ラベルマップ
 */
export const MAINTENANCE_CATEGORY_LABELS: Record<MaintenanceCategory, string> = {
  [MAINTENANCE_CATEGORY.ENGINE]: "エンジン",
  [MAINTENANCE_CATEGORY.INTAKE_EXHAUST]: "吸排気",
  [MAINTENANCE_CATEGORY.COOLING]: "冷却",
  [MAINTENANCE_CATEGORY.DRIVETRAIN]: "駆動系",
  [MAINTENANCE_CATEGORY.SUSPENSION]: "足回り",
  [MAINTENANCE_CATEGORY.BRAKE]: "ブレーキ",
  [MAINTENANCE_CATEGORY.ELECTRICAL]: "電装",
  [MAINTENANCE_CATEGORY.BODY]: "外装",
  [MAINTENANCE_CATEGORY.INTERIOR]: "内装",
  [MAINTENANCE_CATEGORY.AIR_CONDITIONER]: "エアコン",
  [MAINTENANCE_CATEGORY.BODY_REPAIR]: "板金",
  [MAINTENANCE_CATEGORY.CLEANING]: "洗浄",
  [MAINTENANCE_CATEGORY.OTHER]: "その他",
};

/**
 * 整備カテゴリの選択肢一覧（Select用、表示順はsortOrder相当の定義順）
 */
export const MAINTENANCE_CATEGORY_OPTIONS: {
  value: MaintenanceCategory;
  label: string;
}[] = Object.values(MAINTENANCE_CATEGORY).map((value) => ({
  value,
  label: MAINTENANCE_CATEGORY_LABELS[value],
}));
